oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/MappedJwtClaimSetConverter.java

```diff
@@ -0,0 +1,232 @@
+
+	/**
+	 * Constructs a {@link MappedJwtClaimSetConverter} with the provided arguments
+	 *
+	 * This will completely replace any set of default converters.
+	 *
+	 * @param attributeConverters The {@link Map} of converters to use
+	 */
+	public MappedJwtClaimSetConverter(Map<String, Converter<Object, ?>> attributeConverters) {
+		Assert.notNull(attributeConverters, "attributeConverters cannot be null");
+		this.attributeConverters.putAll(attributeConverters);
+	}
+
+	/**
+	 * Construct a {@link MappedJwtClaimSetConverter}, overriding individual attribute
+	 * converters with the provided {@link Map} of {@link Converter}s.
+	 *
+	 * For example, the following would give an instance that is configured with only the default
+	 * attribute converters:
+	 *
+	 * <pre>
+	 * 	MappedJwtClaimSetConverter.withDefaults(Collections.emptyMap());
+	 * </pre>
+	 *
+	 * Or, the following would supply a custom converter for the subject, leaving the other defaults
+	 * in place:
+	 *
+	 * <pre>
+	 * 	MappedJwtClaimsSetConverter.withDefaults(
+	 * 		Collections.singletonMap(JwtClaimNames.SUB, new UserDetailsServiceJwtSubjectConverter()));
+	 * </pre>
+	 *
+	 * To completely replace the underlying {@link Map} of converters, {@see MappedJwtClaimSetConverter(Map)}.
+	 *
+	 * @param attributeConverters
+	 * @return An instance of {@link MappedJwtClaimSetConverter} that contains the converters provided,
+	 *   plus any defaults that were not overridden.
+	 */
+	public static MappedJwtClaimSetConverter withDefaults
+			(Map<String, Converter<Object, ?>> attributeConverters) {
+		Assert.notNull(attributeConverters, "attributeConverters cannot be null");
+
+		Map<String, Converter<Object, ?>> withDefaults = new HashMap<>();
+		withDefaults.put(JwtClaimNames.AUD, AUDIENCE_CONVERTER);
+		withDefaults.put(JwtClaimNames.EXP, TEMPORAL_CONVERTER);
+		withDefaults.put(JwtClaimNames.IAT, TEMPORAL_CONVERTER);
+		withDefaults.put(JwtClaimNames.ISS, ISSUER_CONVERTER);
+		withDefaults.put(JwtClaimNames.JTI, STRING_CONVERTER);
+		withDefaults.put(JwtClaimNames.NBF, TEMPORAL_CONVERTER);
+		withDefaults.put(JwtClaimNames.SUB, STRING_CONVERTER);
+		withDefaults.putAll(attributeConverters);
+
+		return new MappedJwtClaimSetConverter(withDefaults);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Map<String, Object> convert(Map<String, Object> claims) {
+		Assert.notNull(claims, "claims cannot be null");
+
+		Map<String, Object> mappedClaims = new HashMap<>(claims);
+
+		for (Map.Entry<String, Converter<Object, ?>> entry : this.attributeConverters.entrySet()) {
+			Converter<Object, ?> converter = this.attributeConverters.get(entry.getKey());
+			if (converter != null) {
+				Object claim = claims.get(entry.getKey());
+				Object mappedClaim = converter.convert(claim);
+				put(mappedClaims, entry.getKey(), mappedClaim);
+			}
+		}
+
+		Instant issuedAt = (Instant) mappedClaims.get(JwtClaimNames.IAT);
```

Comment:
Separate ticket maybe, but I like the idea of also having IssuedAtDefaultingClaimSetConverter which delegates to MappedJwtSetConverter and performs these steps on the map, e.g.:

```java
public class IssuedAtDefaultingClaimSetConverter implements Converter<...> {
    private final Conveter<...> delegate = MappedJwtClaimSetConverter.withDefaults(...);

    public Map<String, Object> convert(Map<String, Object> claims) {
        Map<String, Object> mappedClaims = this.delegate.convert(claims);

        Instant issuedAt = (Instant) mappedClaims.get(JwtClaimNames.IAT);
        Instant expiresAt = (Instant) mappedClaims.get(JwtClaimNames.EXP);
        if (issuedAt == null && expiresAt != null) {
            mappedClaims.put(JwtClaimNames.IAT, expiresAt.minusSeconds(1));
        }

        return mappedClaims;
}
```

This way, users can easily do

```java
Converter<...> converter = MappedJwtClaimSetConverter.withDefaults(...);
NimbusJwtDecoder#setClaimSetConverter(converter);
```